File: proj8/proj_8_testers/c_files/test5.c
// Test the ability to pass-by-value.
#include <stdio.h>

void foo1(int x) {
  printf("foo1: %d\n", x);
  x = 2*x;
  printf("foo1: %d\n", x);
}

void foo2(int x, int y) {
  printf("foo2: %d\t%d\t%d\n", x, y, x+y);
}

void foo3(char* s) {
  printf("foo3: %s\n", s);
}



int main() {
  foo1(100);

  int a = 10;
  foo1(a);
  printf("main: %d\n", a);

  int x = 10;
  foo1(x);
  printf("main: %d\n", x);

  foo2(a,x);
  printf("main: %d\t%d\n", a, x);

  char* s = "A Lannister always pays his debts.";
  foo3(s);
  printf("main: %s\n", s);

  return 0;
}

File: archive/proj8/proj_8_testers/tests/test5.ml
# Tests the ability to pass-by-value

begin
    foo1(int x) begin
        write("foo1: ", x, "\n");
        x := 2 * x;
        write("foo1: ", x, "\n");
    end

    foo2(int x, int y) begin
        write("foo2: ", x, "\t", y, "\t", x + y, "\n");
    end

    foo3(string s) begin
        write("foo3: ", s, "\n");
    end

    foo1(100)

    int a := 10;
    foo1(a)
    write("main: ", a, "\n");

    int x := 10;
    foo1(x)
    write("main: ", x, "\n");

    foo2(a,x)
    write("main: ", a, "\t", x, "\n");

    string s := "A Lannister always pays his debts.";
    foo3(s)
    write("main: ", s, "\n");
end
